Mage.Sets/src/mage/cards/v/Vault101BirthdayParty.java

```diff
@@ -0,0 +1,119 @@
+        SagaAbility sagaAbility = new SagaAbility(this);
+
+        // I -- Create a 1/1 white Human Soldier creature token and a Food token.
+        sagaAbility.addChapterEffect(this, SagaChapter.CHAPTER_I,
+                new CreateTokenEffect(new HumanWarriorToken()),
+                new CreateTokenEffect(new FoodToken()));
+
+        // II, III -- You may put an Aura or Equipment card from your hand or graveyard onto the battlefield.
+        // If an Equipment is put onto the battlefield this way, you may attach it to a creature you control.
+        sagaAbility.addChapterEffect(this, SagaChapter.CHAPTER_II, SagaChapter.CHAPTER_III,
+                new Vault101BirthdayPartyEffect());
+
+        this.addAbility(sagaAbility);
+
+    }
+
+    private Vault101BirthdayParty(final Vault101BirthdayParty card) {
+        super(card);
+    }
+
+    @Override
+    public Vault101BirthdayParty copy() {
+        return new Vault101BirthdayParty(this);
+    }
+
+}
+
+class Vault101BirthdayPartyEffect extends OneShotEffect {
+
+    private static final FilterCard filter = new FilterCard("");
+
+    static {
+        filter.add(Predicates.or(
+                SubType.AURA.getPredicate(),
+                SubType.EQUIPMENT.getPredicate()
+        ));
+    }
+
+    public Vault101BirthdayPartyEffect() {
+        super(Outcome.PutCardInPlay);
+        this.staticText = "put an Aura or Equipment card from your hand or graveyard onto the battlefield. "
+                         + "If an Equipment is put onto the battlefield this way, you may attach it to a creature you control." ;
+    }
+
+    private Vault101BirthdayPartyEffect(final mage.cards.v.Vault101BirthdayPartyEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public mage.cards.v.Vault101BirthdayPartyEffect copy() {
+        return new mage.cards.v.Vault101BirthdayPartyEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player player = game.getPlayer(source.getControllerId());
+        if (player == null) {
+            return false;
+        }
+        Cards cards = new CardsImpl();
+        cards.addAllCards(player.getHand().getCards(filter, game));
+        cards.addAllCards(player.getGraveyard().getCards(filter, game));
+        TargetCard targetCard = new TargetCard(Zone.ALL, filter);
+        player.choose(outcome, cards, targetCard, source, game);
+        Card card = game.getCard(targetCard.getFirstTarget());
+        if (card == null) {
+            return false;
+        }
+        player.moveCards(card, Zone.BATTLEFIELD, source, game);
+        cards.removeIf(uuid -> game.getState().getZone(uuid) != Zone.LIBRARY);
+        Permanent equipment = game.getPermanent(card.getId());
+        if (equipment == null || !equipment.hasSubtype(SubType.EQUIPMENT, game)) {
+            return player.putCardsOnBottomOfLibrary(cards, game, source, false);

```

Comment:
This doesn't seem right- I don't see anything in the card text about putting cards on bottom of library.

Reply:
You're absolutely right, I based it on Armored Skyhunter and Danitha, Benalia's Hope, and forgot to remove some of the irrelevant code.